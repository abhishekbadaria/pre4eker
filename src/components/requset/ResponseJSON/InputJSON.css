.json {
  margin-bottom: .5rem;
}