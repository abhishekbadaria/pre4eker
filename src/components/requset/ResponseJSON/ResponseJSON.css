.editWrrapper {
  height: calc(100% - 37.6px);
}