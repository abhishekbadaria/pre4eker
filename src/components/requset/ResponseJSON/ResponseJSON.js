import React, { useState } from "react";
import PropTypes from "prop-types";

import AceEditor from "react-ace";

import "ace-builds/src-noconflict/mode-json";
import "ace-builds/src-noconflict/theme-xcode";
import "ace-builds/webpack-resolver";

import Button from "src/components/Button/Button";

const ResponseJSON = ({ responseJSON, setJSON }) => {
  const [body, setBody] = useState(JSON.stringify(responseJSON, null, 1));

  return (
    <div>
      <AceEditor
        name="ResponseBody"
        mode="json"
        theme="xcode"
        value={body}
        onChange={(value) => setBody(value)}
        height="68vh"
        width="100%"
        wrapEnabled
        placeholder="Тело ответа"
        highlightActiveLine={false}
        setOptions={{
          showLineNumbers: false,
          tabSize: 2,
        }}
      />
      <Button
        action={() => setJSON(JSON.parse(body))}
        type="TEXT"
        size="sizeFull"
      >
        Показать
      </Button>
    </div>
  );
};

ResponseJSON.defaultProps = {
  responseJSON: {},
};

ResponseJSON.propTypes = {
  responseJSON: PropTypes.object,
  setJSON: PropTypes.func.isRequired,
};

export default ResponseJSON;
