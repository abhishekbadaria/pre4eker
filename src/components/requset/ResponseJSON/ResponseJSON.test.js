import React from "react";
import { render, fireEvent } from "@testing-library/react";
import ResponseJSON from "./ResponseJSON";

describe("Form", () => {
  it("should handle click on btn", () => {
    const setJSON = jest.fn();

    const { asFragment } = render(<ResponseJSON setJSON={setJSON} />);

    expect(asFragment(<ResponseJSON />)).toMatchSnapshot();
  });
});
