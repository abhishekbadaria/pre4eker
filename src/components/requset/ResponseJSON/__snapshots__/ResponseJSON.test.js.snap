// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should handle click on btn 1`] = `
<DocumentFragment>
  <div
    class="loading"
  >
    <div
      class="loader loader-undefined loader-active"
    >
      <div
        class="loader-inner ball-scale-ripple-multiple"
      >
        <div />
        <div />
        <div />
      </div>
    </div>
  </div>
  <button
    class="sizeFull button normal false basic"
    type="button"
  >
    Показать
  </button>
</DocumentFragment>
`;
